bigquery_etl/operational_monitoring/__init__.py

```diff
@@ -0,0 +1,177 @@
+        for item in query[item_type]
+    ]
+
+
+def _write_sql(project, dataset, slug, kwargs, init):
+    basename = INIT_FILENAME if init else QUERY_FILENAME
+    write_sql(
+        OUTPUT_DIR,
+        f"{project}.{dataset}.{slug}",
+        basename,
+        render(
+            basename,
+            template_folder="operational_monitoring",
+            **kwargs,
+            init=init,
+        ),
+    )
+
+
+def _query_up_to_date(dataset, slug, basename, project_last_modified):
+    """Check whether the project file was updated after the query.sql file was written."""
+    query_path = Path(os.path.join(OUTPUT_DIR, dataset, slug, basename))
+    query_last_modified = datetime.utcfromtimestamp(query_path.stat().st_mtime)
+
+    if not os.path.exists(query_path) or (
+        os.path.exists(query_path)
+        and query_last_modified < project_last_modified.replace(tzinfo=None)
+    ):
+        return False
+
+    return True
+
+
+def _generate_sql(project, dataset):
+    render_kwargs = {
+        "header": "-- Generated via bigquery_etl.operational_monitoring\n",
+        "gcp_project": project,
+        "dataset": dataset,
+    }
+
+    client = storage.Client(project)
+    bucket = client.get_bucket(BUCKET_NAME)
+
+    probes, _ = _download_json_file(project, bucket, "probes.json")
+    dimensions, _ = _download_json_file(project, bucket, "dimensions.json")
+
+    # Iterating over all defined operational monitoring projects
+    for blob in bucket.list_blobs(prefix=PROJECTS_FOLDER):
+        # The folder itself is not a project file
+        if blob.name == PROJECTS_FOLDER:
+            continue
+
+        om_project, project_last_modified = _download_json_file(
+            project, bucket, blob.name
+        )
+        slug = om_project["slug"]
+        render_kwargs.update(
+            {"slug": om_project["slug"], "branches": om_project["branches"]}
+        )
+        if _query_up_to_date(dataset, slug, INIT_FILENAME, project_last_modified):
+            logging.info(
+                f"Queries for {slug} are up to date and will not be regenerated"
+            )
+            continue
+
+        render_kwargs["slug"] = slug
+
+        # Iterating over each dataset to query for a given project.
+        for query in om_project["analysis"]:
+            render_kwargs.update(
+                {
+                    "source": query["source"],
+                    "dimensions": _get_name_and_sql(query, dimensions, "dimensions"),
+                    "probes": _get_name_and_sql(query, probes, "probes"),
+                }
+            )
+
+            _write_sql(project, dataset, slug, render_kwargs, init=True)
+            _write_sql(project, dataset, slug, render_kwargs, init=False)
+
+
+def _run_sql(project, dataset, submission_date):
+    bq_client = bigquery.Client(project=PROD_PROJECT)
+
+    for slug in os.listdir(os.path.join(OUTPUT_DIR, dataset)):
+        destination_table = f"{project}.{dataset}.{slug}"
```

Comment:
For the `slug` part, generally the naming convention for tables is not to use dashes but to use underscores instead. This is how we do it in jetstream: https://github.com/mozilla/jetstream/blob/29efddb0825678b55504ccc07bfe86c4363bdad9/jetstream/__init__.py#L22